### PR TITLE
Branching process added due to the addition of superadmin.

### PR DESCRIPTION
**Pull request by kkana6624**
# Related Issues
#22 Introduce the concept of super administrator

# Changes
* Add new migration file
* AdminLive with additional processing by adding superadmin

# Question
* I assume from the Issue that the general administrator can update himself but not other administrators.
Is that Ok?

# Review Perspective
* Whether the content of the implementation is appropriate
* Whether UI behaviour is appropriate
* Whether the UI display (superadmin) is as expected
* Whether the format and title of the pull request are appropriate

add_superadmin

Sign in with taro(superadmin)
image

Sign in with jiro
image

### CODE REVIEW

**Comment by kuroda** [comment]
@kkana6624 

> I assume from the Issue that the general administrator can update himself but not other administrators. Is that Ok?

No. A non-super administartor cannot update oneself.

I plan to provide a separate UI for administrators to change their own information.

File: lib/live_web_server_web/live/admin_live/administrator.html.heex [hunk 1]
```diff
@@ -1,8 +1,14 @@
...
+  <%= cond do %>
+    <%= @administrator.superadmin -> %>
+      <td>
+        <%= @administrator.username %>(superadmin)
+      </td>
+    <% true -> %>
+      <td>
+        <%= @administrator.username %>
+      </td>
+  <% end %>
```
Comment (kuroda) [inline lib/live_web_server_web/live/admin_live/administrator.html.heex]: Here, You'd be better off writing it like this:

```html
  <td>
    <%= @administrator.username %> 
    <%= if @administrator.superadmin %>(superadmin)<% end%>
  </td>
```

File: priv/repo/migrations/20240910011342_add_super_admin.exs [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+
+    alter table(:core_administrators, primary_key: false) do
+      add(:superadmin, :boolean, null: false, default: false)
+    end
+
+    alter table(:core_deleted_administrators, primary_key: false) do
+      add(:superadmin, :boolean, null: false, default: false)
+    end
+  end
+end
```
Comment (kuroda) [inline priv/repo/migrations/20240910011342_add_super_admin.exs]: Until Version 1 of this software is released, please do not create migration scripts that perform `ALTER` operations.

Instead, rewrite your existing migration scripts _directly_.

Also, the `superadmin` column is not required for `active_administrators` and `deleted_administrators`.

**Review by kuroda — CHANGES_REQUESTED** [review]
Please read my comments and correct the source code.

File: lib/live_web_server_web/live/admin_live/administrator.html.heex (hunk 1 above)
Comment (kkana6624) [reply lib/live_web_server_web/live/admin_live/administrator.html.heex]: I ran the code you suggested in my environment and it did not work as expected, so I modified it as follows.
```html 
  <td>
    <%= @administrator.username %>
    <%= if @administrator.superadmin, do: "(superadmin)" %>
  </td>
```

The UI seems to be working as expected, but please let us know if there are any problems.

File: priv/repo/migrations/20240910011342_add_super_admin.exs (hunk 2 above)
Comment (kkana6624) [reply priv/repo/migrations/20240910011342_add_super_admin.exs]: Unnecessary migration files were deleted and appended to existing migration files.

Also, unnecessary edits have been removed.

**Comment by kkana6624** [comment]
@kuroda 
I have corrected the points that you pointed out. Please confirm.

**Comment by kuroda** [comment]
@kkana6624 

Thank you.

The `btn-danger` class is specified for the “Delete” button in the admin table. This class is not in daisyUI. Use `btn-neutral`.

Also, although not included in the initial instructions, please address the following additional points:

* If a user is logged in as a non-super administrator, disable the "Undelete" button in the deleted administrators table.

**Comment by kkana6624** [comment]
@kuroda 
Incorrect CSS class designations have been corrected.

Additional requirements have also been implemented.

If the user is a superadmin
image

If the user is not a superadmin
image

**Comment by kkana6624** [comment]
Although not related to this pull request, 
it was slightly disconcerting that the screen did not show who was logged in.

Is it because the users themselves know who is signed in and therefore it is unnecessary?

**Comment by kuroda** [comment]
@kkana6624 

I have confirmed that all behavior is correct. Thank you very much.

Now, your PR contains two commits. I believe that a PR of this size should be submitted as a single commit.

So, please use the `git rebase -i origin/main` command to squash the commits into one.

The message of that single commit should be in the following format:

```
Introduce the concept of super administrator

* Fixed a migration file, a schema file, and a seeding script
* Disabled some buttons for non-super administrators
```

**Comment by kuroda** [comment]
@kkana6624 

> it was slightly disconcerting that the screen did not show who was logged in.

You are right.

I opened a new issue #26 to address this inconvenience and add a new feature.

**Comment by kkana6624** [comment]
@kuroda 
Thank you for your review.
I have aggregate the commits and re-pushed.

**Review by kuroda — APPROVED** [review]

**Pull request merged**